spark/v3.4/spark/src/test/java/org/apache/iceberg/TaskCheckHelper.java

```diff
@@ -45,61 +47,58 @@ public static void assertEquals(FileScanTask expected, FileScanTask actual) {
-    Assert.assertEquals(
-        "Should match the serialized record size",
-        expected.fileSizeInBytes(),
-        actual.fileSizeInBytes());
-    Assert.assertEquals(
-        "Should match the serialized record value counts",
-        expected.valueCounts(),
-        actual.valueCounts());
-    Assert.assertEquals(
-        "Should match the serialized record null value counts",
-        expected.nullValueCounts(),
-        actual.nullValueCounts());
-    Assert.assertEquals(
-        "Should match the serialized record lower bounds",
-        expected.lowerBounds(),
-        actual.lowerBounds());
-    Assert.assertEquals(
-        "Should match the serialized record upper bounds",
-        expected.upperBounds(),
-        actual.upperBounds());
-    Assert.assertEquals(
-        "Should match the serialized record key metadata",
-        expected.keyMetadata(),
-        actual.keyMetadata());
-    Assert.assertEquals(
-        "Should match the serialized record offsets",
-        expected.splitOffsets(),
-        actual.splitOffsets());
-    Assert.assertEquals(
-        "Should match the serialized record offsets", expected.keyMetadata(), actual.keyMetadata());
+    assertThat(actual.location())
+        .as("Should match the serialized record path")
+        .isEqualTo(expected.location());
+    assertThat(actual.format())
+        .as("Should match the serialized record format")
+        .isEqualTo(expected.format());
+    assertThat(actual.partition().get(0, Object.class))
+        .as("Should match the serialized record partition")
+        .isEqualTo(expected.partition().get(0, Object.class));
+    assertThat(actual.recordCount())
+        .as("Should match the serialized record count")
+        .isEqualTo(expected.recordCount());
+    assertThat(actual.fileSizeInBytes())
+        .as("Should match the serialized record size")
+        .isEqualTo(expected.fileSizeInBytes());
+    assertThat(actual.valueCounts())
+        .as("Should match the serialized record value counts")
+        .isEqualTo(expected.valueCounts());
+    assertThat(actual.nullValueCounts())
+        .as("Should match the serialized record null value counts")
+        .isEqualTo(expected.nullValueCounts());
+    assertThat(actual.lowerBounds())
+        .as("Should match the serialized record lower bounds")
+        .isEqualTo(expected.lowerBounds());
+    assertThat(actual.upperBounds())
+        .as("Should match the serialized record upper bounds")
+        .isEqualTo(expected.upperBounds());
+    assertThat(actual.keyMetadata())
+        .as("Should match the serialized record key metadata")
+        .isEqualTo(expected.keyMetadata());
+    assertThat(actual.splitOffsets())
+        .as("Should match the serialized record offsets")
+        .isEqualTo(expected.splitOffsets());
+    assertThat(actual.keyMetadata())
+        .as("Should match the serialized record offsets")
+        .isEqualTo(expected.keyMetadata());
```

Comment:
The `keyMetadata()` method is already verified at L93-95. It's a pre-existing mistake, though.

Reply:
Thanks for checking this! The Spark 3.5 alos has this duplicated check, let me remove them from both versions.